cmd/clusterctl/internal/test/fake_objects.go

```diff
@@ -1021,18 +1021,22 @@ func (f *FakeClusterResourceSet) Objs() []runtime.Object {
 	}
 
 	// Ensures all the binding with the clusters where resources are applied.
-	if len(f.clusters) > 0 {
+	for _, cluster := range f.clusters {
 		binding := &addonsv1alpha3.ClusterResourceSetBinding{
 			TypeMeta: metav1.TypeMeta{
 				Kind:       "ClusterResourceSetBinding",
 				APIVersion: addonsv1alpha3.GroupVersion.String(),
 			},
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      f.name,
-				Namespace: f.namespace,
+				Name:      cluster.Name,
+				Namespace: cluster.Namespace,
```

Comment:
Another thing
Given this change, I would expect 
"addons.cluster.x-k8s.io/v1alpha3, Kind=ClusterResourceSetBinding, ns1/crs1" --> "addons.cluster.x-k8s.io/v1alpha3, Kind=ClusterResourceSetBinding, ns1/cluster1"
Also in objectgraph_test.go

Reply:
You're right, fixed them.